### PR TITLE
VM initialization and creating contracts

### LINKED ISSUE #15: Handle Wasm VM Initialization (the CREATE flag)
NOTE: This issue doesn't include function selectors, which will be handled separately.

### PR DESCRIPTION
**Pull request by virgil-serbanuta**
Fixes #15

### CODE REVIEW

File: pykwasm/src/pykwasm/kdist/wasm-semantics/ulm-wasm.md
Comment (sskeirik) [inline pykwasm/src/pykwasm/kdist/wasm-semantics/ulm-wasm.md]: Just curious: is this an optimization or is it required because it is not covered by another case? I don't have a strong opinion about it, just curious.

File: pykwasm/src/pykwasm/kdist/wasm-semantics/ulm-wasm.md
Comment (sskeirik) [inline pykwasm/src/pykwasm/kdist/wasm-semantics/ulm-wasm.md:208]: This is a style question, so feel free to resolve it however. On some other added lines, the two arrow styles (=> and ~>) have matching columns. Here, they're different. I'm not sure if this is intended or if, e.g., some editor is mixing whitespace styles.

File: pykwasm/src/pykwasm/kdist/wasm-semantics/ulm-wasm.md
Comment (sskeirik) [inline pykwasm/src/pykwasm/kdist/wasm-semantics/ulm-wasm.md:330]: A similar style comment about arrows as above.

**Review by sskeirik — APPROVED** [review]
Looks good, thanks! I have a few minor questions/comments.

Also, after clicking the merge button but before merging, could we change the generated commit header to be a bit more descriptive? To me, at the moment, the PR title (which is used to auto-populate the commit header), is a bit more broad than the PR contents.

File: pykwasm/src/pykwasm/kdist/wasm-semantics/ulm-wasm.md
Comment (virgil-serbanuta) [reply pykwasm/src/pykwasm/kdist/wasm-semantics/ulm-wasm.md:208]: I'm not sure if I understand this properly, but I went through all occurrences of `~>` in this file and I think that it works like this:
1. Here (and in the place below with a similar comment), there are three items in a K list on the RHS of `=>` (setContractModIdx, resolveCurModuleFuncExport, setSuccessStatus). Because the K list made of these three is a child of the rewrite, they are indented.
2. In other places, we have the rewrite as a child of `~>` (e.g. the rule at line 136) and there I'm using the following pattern: an operator-like thing (e.g. `~>` and `=>`) are printed like this:
```
term
op term
op term
...
```
This means that something like `(a => b) ~> c` would be formatted as one of:
```
(a => b)
~> c
```
or
```
(a
=> b
)
~> c
```
Arguably, the latter may work better as:
```
(   a
    => b
)
~> c
```
(with or without spaces before a) since the rewrite is a child of the parenthesis. Is this what you meant? If yes, I can just fix this. (I can also fix it if the issue is different, it's just that I'm not fully sure what you think should be fixed).

File: pykwasm/src/pykwasm/kdist/wasm-semantics/ulm-wasm.md
Comment (virgil-serbanuta) [reply pykwasm/src/pykwasm/kdist/wasm-semantics/ulm-wasm.md]: I thought it was a corner case that should be handled separately, but it wasn't. I included it in the rule below.

File: pykwasm/src/pykwasm/kdist/wasm-semantics/ulm-wasm.md
Comment (sskeirik) [reply pykwasm/src/pykwasm/kdist/wasm-semantics/ulm-wasm.md:208]: Hmm... I haven't thought about this to the level of detail that you have! Since this is just a style question anyway, let's keep this as is and we can discuss later the possibility of a K coding style.

**Pull request merged**
